### PR TITLE
Remove p2p peer ID from publisher multiaddr given to sync

### LINKED ISSUE #865: Manual sync from previously unknown peers fail with `no good addresses`
When a provider over datatransfer is completely unknown to the indexer, manual sync seems to fail with `no good addresses` error. This is because the datatransfer libp2p host peerstore is not updated with the address from the explicit sync and therefore the host fails to connect.

Update the libp2p peerstore with the address provided via the sync command before manual sync.

### PR DESCRIPTION
**Pull request by gammazero**
Sync does not work if the publisher multiaddr contains a p2p peer ID. This PR detects if there is one and truncates the multiaddress at the start of the /p2p/<peerID> part of the multiaddr.

This allows the sync command to work whether or not the specified publisher multiaddress has a p2p peer ID.

Fixes #865

### CODE REVIEW

**Comment by ischasny** [comment]
@gammazero the issue surfaced up when querying root CID, i.e. here: https://github.com/ipni/storetheindex/blob/main/dagsync/p2p/protocol/head/head.go#L85. Would we need to strip out p2p ID in multiple places then?

**Comment by masih** [comment]
For reference I have captured [this](https://github.com/libp2p/go-libp2p/issues/2001) issue to ask libp2p folks if they think this is expected behaviour.

File: dagsync/subscriber.go [hunk 1]
```diff
@@ -406,7 +407,7 @@ func (s *Subscriber) Sync(ctx context.Context, peerID peer.ID, nextCid cid.Cid,
 
 	var peerAddrs []multiaddr.Multiaddr
 	if peerAddr != nil {
-		peerAddrs = []multiaddr.Multiaddr{peerAddr}
+		peerAddrs = []multiaddr.Multiaddr{rmPeerIDFromMultiaddr(peerAddr)}
```
Comment (masih) [inline dagsync/subscriber.go:410]: The code would be much less verbose if we use `peer.SplitAddr`; e.g.:

```suggestion
		transport, _ := peer.SplitAddr(peerAddr)
		peerAddrs = []multiaddr.Multiaddr{rmPeerIDFromMultiaddr(transport)}
```

**Review by masih — CHANGES_REQUESTED** [review]
LGTM, thank you for figuring this out Andrew 👍 

Left a comment to use `peer.SplitAddr` instead which should relax the need for string manipulation.

**Comment by masih** [comment]
Fixes #865

File: dagsync/subscriber.go (hunk 1 above)
Comment (gammazero) [reply dagsync/subscriber.go:410]: Agree, will update.

**Comment by gammazero** [comment]
@ischasny Potentially we would need to do this in any place that the indexer dials another host.  Other than CLI use, This is really only done during sync, so I think we are OK. This should all go away completely when there is a fix for the [libp2p issue](https://github.com/libp2p/go-libp2p/issues/2001) that was filed.

**Pull request merged**